datafusion/physical-expr-adapter/src/schema_rewriter.rs

```diff
@@ -220,13 +223,111 @@ impl<'a> DefaultPhysicalExprAdapterRewriter<'a> {
         &self,
         expr: Arc<dyn PhysicalExpr>,
     ) -> Result<Transformed<Arc<dyn PhysicalExpr>>> {
+        if let Some(transformed) = self.try_rewrite_struct_field_access(&expr)? {
+            return Ok(Transformed::yes(transformed));
+        }
+
         if let Some(column) = expr.as_any().downcast_ref::<Column>() {
             return self.rewrite_column(Arc::clone(&expr), column);
         }
 
         Ok(Transformed::no(expr))
     }
 
+    fn try_rewrite_struct_field_access(
+        &self,
+        expr: &Arc<dyn PhysicalExpr>,
+    ) -> Result<Option<Arc<dyn PhysicalExpr>>> {
+        let get_field_expr = match expr.as_any().downcast_ref::<ScalarFunctionExpr>() {
+            Some(expr) => expr,
+            None => return Ok(None),
+        };
+
+        if get_field_expr.name() != "get_field" {
+            return Ok(None);
+        }
+
+        if get_field_expr
+            .fun()
+            .inner()
+            .as_any()
+            .downcast_ref::<GetFieldFunc>()
+            .is_none()
+        {
+            return Ok(None);
+        }
+
+        let source_expr = match get_field_expr.args().first() {
+            Some(expr) => expr,
+            None => return Ok(None),
+        };
+
+        let field_name_expr = match get_field_expr.args().get(1) {
+            Some(expr) => expr,
+            None => return Ok(None),
+        };
+
+        let lit = match field_name_expr
+            .as_any()
+            .downcast_ref::<expressions::Literal>()
+        {
+            Some(lit) => lit,
+            None => return Ok(None),
+        };
+
+        let field_name = match lit.value() {
```

Comment:
try_as_str might be useful here: https://docs.rs/datafusion/latest/datafusion/common/enum.ScalarValue.html#method.try_as_str